src/ams.py

```diff
@@ -135,17 +135,14 @@ def install(self):
         """Install AMS including its Snap."""
         try:
             res = self._charm.model.resources.fetch("ams-snap")
+            # FIXME: Install the ams snap from a resource until we make the
+            # snaps in the snap store unlisted
+            if res is not None and res.stat().st_size:
```

Comment:
Should we raise an exception instead of continuing if no 'ams-snap' resource is provided or if a dummy file is provided?